In [29]:
a = "Hello Python"
print(a[6:]+" "+a[:5])

Python Hello


### Slicing : 문자의 인덱스를 통해 단어를 추출하는 것
+ a 에 hello Python이 저장되어있다.
+ Python을 먼저 추출해야하므로 a[6:12]로 작성한다
+ 후에 Hello 추출을 위해 a[0:5]기입한 후 +로 묶어준다.

In [19]:
b = "Hello Python World"
b[13:] + " " + b[6:12] + " " + b[0:5]

'World Python Hello'

+ 위에 문제와 흡사하며 단락이 3개로 늘어났다

In [18]:
c = "Hello"
c[::-1]

'olleH'

+ 슬라이싱에서 x:y를 통해 x~y범위를 정했다면 x:y:z에서 z는 step을 의미한다.
+ ::2라면 2칸씩 넘지만 -1을 사용하면 앞과뒤에 인덱스차가 -1이 되어야하므로 앞에것에서 뒤에것을 뺀 -1 즉 문자열이 반대로 출력이 된다.

In [21]:
s = "python"
s[0],s[0][0].s[0][0][0]

AttributeError: 'str' object has no attribute 's'

+ s[0]은 python에서 0번재 인덱스에 있는 문자'p'가 나오지만 그 외 s[0][0]과 s[0][0][0]은 2차원 3차원 배열의 인덱스를 말하기때문에 값이 나오지 않는 것이다.

In [23]:
s[-100], s[100]

IndexError: string index out of range

+ 인덱스의 범위가 python의 길이인데 -100과 100은 초과했으므로 출력되지않는다.

In [51]:
s[-100,100]

TypeError: string indices must be integers

+ 이 또한 2차월 배열에서 -100,100의 인덱스위치인데 s는 1차원이기에 출력되지않는다.

In [52]:
s[1:-1]

'ytho'

+ 'python'에서 인덱스 1 부터 끝자리 5자리에서 -1한 부분까지 출력을 하라는 뜻이다.

In [53]:
for i in range(1,100,2): 
    print(i,end = " ")

1 3 5 7 9 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 51 53 55 57 59 61 63 65 67 69 71 73 75 77 79 81 83 85 87 89 91 93 95 97 99 

+ for문을 통해 반복하되 range(1,100,2) 즉 범위1에서 100까지 2씩 뛰어넘어서 출력하란 뜻이다.

In [2]:
result = 0
j = 0
while j<=100 :
    if j%2==0 :
        result=result + j
    j=j+1
print(result)

2550


+ while << j가 100이 될때까지 반복
+ if << j가 2로 나눠지면 result값에 j를 더함
+ j=j+1 << while 안에서 j를 1씩 증가시킴

In [59]:
num=input()
print(num[::-1])

3125
5213


+ input()으로는 문자열형태로 받는다. 고로 num에 3125란 문자열이 입력되었기에 num(::-1)로 역출력한다.

In [1]:
prime=int(input())
for i in range(1,prime+1) :
    tmp = True
    for j in range(2,i):
        if i%j==0:
            tmp=False
            break
    if tmp:
        print(i,end=" ")

50
1 2 3 5 7 11 13 17 19 23 29 31 37 41 43 47 

+ 3번째 줄에서 for문안에서 tmp를 true로 초기화
+ 5번째 줄에서 나눈값이 0이 하나라도 된다면 False값으로 하고 반복문 탈출
+ 8번째 줄에서 위에 for에서 합당한 값이 없다면 tmp는 계속 true임으로 출력, 즉 소수 출력

In [77]:
star=int(input())
space=0
while star>0:
    print(' '*space + "*"*star)
    star-=1
    space+=1

5
*****
 ****
  ***
   **
    *


+ while << 위에서 입력받은 star가 0이 될때까지 반복
+ star와 space를 통하여 print에서 갯수 조절을 해준다.

In [20]:
line = input()
count = 0;
for i in range(0,len(line)):
    if count==10:
        print("")
        count = 0
    print(line[i],end='')
    count = count+1

OneTwoThreeFourFiveSixSevenEightNineTen
OneTwoThre
eFourFiveS
ixSevenEig
htNineTen

+ for << 0에서 len(line) line의 문자길이까지 반복한다.
+ if << i번이 10번째가 될때마다 한줄씩 띄운다
+ print로 line[0]부터 line[9]까지 반복한다.

In [4]:
sugar = int(input())
if sugar == 4 or sugar == 7 :
    print("-1")
elif sugar % 5 == 0 :
    print(int(sugar/5))
elif (sugar%5)%2 == 1 :
    print(int(sugar/5+1))
else :
    print(int((sugar/5)+2))

15
3


+ 설탕이 4와 7일땐 불가능하므로 -1을 출력
+ 첫번째 elif << 5로 나눠떨어지면 5kg설탕봉지만 필요하므로 그 갯수를 출력
+ 두번째 elif << 5로 나눴을때 남은 나머지가 2로 다시 나눠서 1이 된다면 총갯수는 5kg갯수에 1개 더 추가한다.
+ else 나머지가 2 혹은 4가 된다면 5kg갯수에 2개를 더 추가한다.

![](https://raw.githubusercontent.com/Lee-Kyu-Young/Harlequeen/master/answer.PNG)

과제를 하며 거의 처음부터 다시 공부를 하였다. 지난 인강은 전부 다시 들어야했고 구글과 강의자료는 계속 끼고 있어야했다. 하지만 스스로 실력항상에 도움이 되었다 생각하고 아직 안배운 elif문까지 예습하게 되어서 매우 알찬 첫 과제였다고 생각한다.